examples/launch_ask_tell_scheduler.py

```diff
@@ -0,0 +1,185 @@
+        )
+
+    def save(self, output_path: Optional[Union[str, Path]] = None) -> Path:
+        """
+        Save the scheduler to a given path
+        :param output_path: Complete output path for the scheduler,
+            defaults to ask_tell_scheduler-after-trial={self.trial_counter}.dill
+        :return: output path
+        """
+        if output_path is None:
+            output_path = f"ask_tell_scheduler-after-trial={self.trial_counter}.dill"
+
+        with open(output_path, "wb") as f:
+            dill.dump(self, f)
+        return output_path
+
+    @staticmethod
+    def from_file(output_path: Union[str, Path]) -> "AskTellScheduler":
+        """
+        Load the scheduler from path
+        :param output_path: Complete output path for the scheduler
+        """
+        with open(output_path, "rb") as f:
+            scheduler = dill.load(f)
+        return scheduler
+
+
+def target_function(x, noise: bool = True):
+    fx = x * x + np.sin(x)
+    if noise:
+        sigma = np.cos(x) ** 2 + 0.01
+        noise = 0.1 * np.random.normal(loc=x, scale=sigma)
+        fx = fx + noise
+
+    return fx
+
+
+def get_objective():
+    metric = "mean_loss"
+    mode = "min"
+    max_iterations = 100
+    config_space = {
+        "x": uniform(-1, 1),
+    }
+    return metric, mode, config_space, max_iterations
+
+
+def inspect_objective():
+    """
+    In this function, we will inspect the objective by plotting the target function
+    :return:
+    """
+    import matplotlib.pyplot as plt
+
+    metric, mode, config_space, max_iterations = get_objective()
+
+    plt.set_cmap("viridis")
+    x = np.linspace(config_space["x"].lower, config_space["x"].upper, 400)
+    fx = target_function(x, noise=False)
+    noise = 0.1 * np.cos(x) ** 2 + 0.01
+
+    plt.plot(x, fx, "r--", label="True value")
+    plt.fill_between(x, fx + noise, fx - noise, alpha=0.2, fc="r")
+    plt.legend()
+    plt.grid()
+    plt.show()
+
+
+def tune_with_random_search() -> TrialResult:
+    metric, mode, config_space, max_iterations = get_objective()
+    scheduler = AskTellScheduler(base_scheduler=RandomSearch(config_space, metric=metric, mode=mode))
+    for iter in range(max_iterations):
+        trial_suggestion = scheduler.ask()
+        test_result = target_function(**trial_suggestion.config)
+        scheduler.tell(trial_suggestion, {metric: test_result})
+    return scheduler.best_trial(metric)
+
+
+def save_restart_with_gp() -> TrialResult:
+    metric, mode, config_space, max_iterations = get_objective()
+    scheduler = AskTellScheduler(base_scheduler=BayesianOptimization(config_space, metric=metric, mode=mode))
+    for iter in range(int(max_iterations / 2)):
+        trial_suggestion = scheduler.ask()
+        test_result = target_function(**trial_suggestion.config)
+        scheduler.tell(trial_suggestion, {metric: test_result})
+
+    scheduler.save("scheduler-checkpoint.dill")
```

Comment:
If we just serialize with dill, couldnt we skip the save/load method and directly dill the scheduler to simplify the example?

Reply:
Good point, I thought making it a function is easier but I buy the shorter=simpler argument so moved it to the function.